target_chains/solana/programs/pyth-solana-receiver/src/lib.rs

```diff
@@ -27,6 +32,51 @@ declare_id!("DvPfMBZJJwKgJsv2WJA8bFwUMn8nFd5Xpioc6foC3rse");
 pub mod pyth_solana_receiver {
     use super::*;
 
+    pub fn initialize(ctx: Context<Initialize>, initial_config: Config) -> Result<()> {
+        let config = &mut ctx.accounts.config;
+        **config = initial_config;
+        Ok(())
+    }
+
+    pub fn request_governance_authority_transfer(
+        ctx: Context<Governance>,
+        target_governance_authority: Pubkey,
+    ) -> Result<()> {
+        let config = &mut ctx.accounts.config;
+        config.target_governance_authority = Some(target_governance_authority);
+        Ok(())
+    }
+
+    pub fn authorize_governance_authority_transfer(
+        ctx: Context<AuthorizeGovernanceAuthorityTransfer>,
+    ) -> Result<()> {
+        let config = &mut ctx.accounts.config;
+        config.governance_authority = config.target_governance_authority.unwrap();
```

Comment:
i know it's checked to be not None on anchor but still scares me a bit.

Reply:
mm worst case it panics?

Reply:
yeah but in this case I recommend using `expect("this is checked in anchor and should never fail")` to make the code more clear.